### ISSUE TITLE
Bound for the skip length in edit lists

### ISSUE THREAD

**Issue by silverdaz** [samtools/hts-specs#506]
It would be wise to specify the possible range for the skip lengths, if an edit list is used.

What we would like to avoid is carrying a segment in the data portion, which is in fact entirely skipped when decrypting, or reencrypting the header for another destination. Of course, we are not interested in _producing_ such files, but the spec doesn't prevent it either. We don't want to be carrying malicious segments, which after edit list manipulations are not ignored anymore.

I am not sure if the following suggestion is enough to cover all cases properly, but I think that a skip length `s` should be bound as follows.

Let `S` be the segment size, and `p` be the current position within any segment.
Let `p0 = p - (p % S)` (ie, the start of the given segment).
Then, `p + s < p0 + 2 * S`.
In other words, for whatever position within a segment `p%S`, we have `p%S + s < 2*S`.
```
|----------------------|-----------------------| (2 segments)
           ô----------- s ------------ô
           |
           p
```
That way, even if `p` is the last byte of a segment, the next byte read must be in the adjacent segment (and we therefore never carry an "unused" segment).
Note: the current position `p` is calculated as the some of all lengths prior to `s` (and I didn't want to overload the issue with sums and subscripts)
